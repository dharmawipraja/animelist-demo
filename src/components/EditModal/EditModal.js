import React, { useEffect } from 'react';
import Backdrop from '@mui/material/Backdrop';
import Box from '@mui/material/Box';
import Stack from '@mui/material/Stack';
import MUIModal from '@mui/material/Modal';
import Fade from '@mui/material/Fade';
import Typography from '@mui/material/Typography';
import TextField from '@mui/material/TextField';
import Button  from '@mui/material/Button';
import { useForm } from 'react-hook-form';

const style = {
  position: 'absolute',
  overflow: 'scroll',
  top: '50%',
  left: '50%',
  transform: 'translate(-50%, -50%)',
  width: '60%',
  height: '60%',
  bgcolor: 'background.paper',
  border: '1px solid #d3d3d3',
  borderRadius: 3,
  boxShadow: 24,
  p: 4
};

const watchedValue = (watch) => {
  const collectionName = watch('collectionNameForm');

  return {
    collectionName
  };
};

function EditModal({ title, isOpen, onClose, onSubmit }) {
  const { register, watch, reset } = useForm();
  const { collectionName } = watchedValue(watch)

  useEffect(() => {
    reset();
  }, [isOpen, reset]);

  const renderAddCollection = () => (
    <form>
      <TextField
        {...register("collectionNameForm", { required: true })}
        id="outlined-basic" 
        label="Collection Name"
        variant="outlined" 
        sx={{ width: '90%' }}
        defaultValue={title}
      />
    </form>

  )

  const onSubmitButton = (onSubmit) => () => {
    onSubmit(collectionName)
  };

  return (
    <div>
      <MUIModal
        aria-labelledby="transition-modal-title"
        aria-describedby="transition-modal-description"
        open={isOpen}
        onClose={onClose}
        closeAfterTransition
        BackdropComponent={Backdrop}
        BackdropProps={{
          timeout: 500,
        }}
      >
        <Fade in={isOpen}>
          <Box sx={style}>
            <Stack spacing={3} sx={{ height: '90%'}} >
              <Typography id="transition-modal-title" variant="h5" component="h2">
                Edit Collection Name
              </Typography>
              {renderAddCollection()}
            </Stack>
            <Box sx={{ display: 'flex', justifyContent: 'flex-end' }} >
              <Button variant="contained" onClick={onSubmitButton(onSubmit)}>Submit</Button>
            </Box>
          </Box>
        </Fade>
      </MUIModal>
    </div>
  );
}

export default EditModal;
