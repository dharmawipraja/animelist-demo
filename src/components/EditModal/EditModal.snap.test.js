import renderer from 'react-test-renderer';
import { useForm } from 'react-hook-form';

import EditModal from './EditModal';

jest
  .mock('react-hook-form', () => ({
    useForm: jest.fn()
  }))
  .mock('@mui/material/Modal', () => 'Modal')

describe('EditModal', () => {
  it('should renders EditModal component', () => {
    const props = {
      title: 'Edit Modal',
      isOpen: true,
      onClose: jest.fn(),
      onSubmit: jest.fn(),
      isCreate: false
    };

    useForm.mockReturnValue({
      register: jest.fn(),
      watch: jest.fn(),
      reset: jest.fn(),
      setError: jest.fn(),
      formState: { errors: {} }
    });

    const tree = renderer.act(() => {
      renderer
        .create(<EditModal {...props} />, { createNodeMock: node => document.createElement(node.type) })
        .toJSON();
    })
    expect(tree).toMatchSnapshot();
  });
});
