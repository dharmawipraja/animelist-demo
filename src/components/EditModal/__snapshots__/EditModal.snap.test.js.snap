// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditModal should renders EditModal component 1`] = `
Object {
  "then": [Function],
}
`;
